lib/kafka_rest_client.js

```diff
@@ -107,13 +110,33 @@ KafkaRestClient.prototype.scheduleTopicMapRefresh = function scheduleTopicMapRef
             self.discoverTopics(function refreshTopicsCallback(err, isDone) {
                 if (err === null && isDone) {
                     self.enable = true;
+                } else {
+                    self.scheduleReconnect(0);
                 }
             });
         }, self.proxyRefreshTime);
         /* eslint-enable no-undef,block-scoped-var */
     }
 };
 
+// Reconnect to rest proxy if scheduleTopicMapRefresh fails
+KafkaRestClient.prototype.scheduleReconnect = function scheduleReconnect(retry) {
+    var self = this;
+    if (retry < reconnectWaitTime.length) {
+        /* eslint-disable no-undef,block-scoped-var */
+        setTimeout(function reconnect() {
+            self.discoverTopics(function refreshTopicsCallback(err, isDone) {
```

Comment:
I wonder if you can refactor this and merge with the above call back function ? (So that we can only define one callback)